script/src/scheduler.rs

```diff
@@ -761,26 +791,40 @@ where
                 .key();
             self.suspend_vm(&id)?;
         }
-
-        let id = self.next_vm_id;
-        self.next_vm_id += 1;
-        let (context, mut machine) = self.create_dummy_vm(&id)?;
-        {
-            let mut sc = context.snapshot2_context().lock().expect("lock");
-            let (program, _) = sc.load_data(data_piece_id, offset, length)?;
-            let metadata = parse_elf::<u64>(&program, machine.machine.version())?;
-            let bytes = machine.load_program_with_metadata(&program, &metadata, args)?;
-            sc.mark_program(&mut machine.machine, &metadata, data_piece_id, offset)?;
-            machine
-                .machine
-                .add_cycles_no_checking(transferred_byte_cycles(bytes))?;
-        }
         self.instantiated.insert(id, (context, machine));
         self.states.insert(id, VmState::Runnable);
 
         Ok(id)
     }
 
+    // Load the program into an empty vm.
+    fn load_vm_program(
+        &mut self,
+        context: &MachineContext<DL>,
+        machine: &mut Machine,
+        data_piece_id: &DataPieceId,
+        offset: u64,
+        length: u64,
+        args: Option<(u64, u64, u64)>,
```

Comment:
How about creating a ne struct:
```rust
struct VmArgs {
  vm_id: u64,
  argc: u64,
  argv: u64,
}
```
to replace the anonymouse turple `(u64, u64, u64)`?  This would improve code readability and make the parameters more self-explanatory.

Reply:
In the previous internal review, Xuejie suggested that I use `Option<(u64, u64, u64)>`. I am not sure whether I should make a struct for them now.

Reply:
I'm not sure either.

Reply:
I did re-check offline communication logs, previously this is represented using the following structure:

```
pub enum BootArgsType {
    Static,
    Stream { vm_id: u64, argc: u64, argp: u64 },
}
```

Since the enum only had 2 variant: one that has 3 arguments, while the other is empty. I just believe that this 2-value enum, could in fact simply be an `Option`.

I have no opinion whether it is `Option<(u64, u64, u64)>` or `Option<ArgvPointer>`, both feel the same to me. I never say `(u64, u64, u64)` is better in readability to `ArgvPointer` or another structure, both feel the same to me but I do understand a different opinion might occur.